samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -33,6 +33,21 @@ static struct bt_conn_le_create_param *create_param = BT_CONN_LE_CREATE_CONN;
 void BLEConnectivityManager::FilterMatch(bt_scan_device_info *device_info, bt_scan_filter_match *filter_match,
 					 bool connectable)
 {
+	if (!filter_match) {
+		LOG_ERR("Filter match error");
+		return;
+	}
+
+	if (filter_match->addr.match) {
+		/* Result comes from the recovery mechanism */
+		memcpy(&Instance().mRecovery.mScannedDevice.mAddr, filter_match->addr.addr,
```

Comment:
nit: it's a matter of preferece, but you could probably just assign the variables: `Instance().mRecovery.mScannedDevice.mAddr = filter_match->addr.addr`.